src/main/java/com/exactpro/th2/readcsv/Main.java

```diff
@@ -95,33 +102,66 @@ public static void main(String[] args) {
                     configuration.getSourceDirectory(),
                     (Path path) -> configuration.getAliases().entrySet().stream()
                             .filter(entry -> entry.getValue().getNameRegexp().matcher(path.getFileName().toString()).matches())
-                            .map(it -> new StreamId(it.getKey(), Direction.FIRST))
+                            .map(it -> new StreamId(it.getKey()))
                             .collect(Collectors.toSet()),
                     files -> files.sort(pathComparator),
                     path -> true
             );
 
             var rootId = commonFactory.getRootEventId();
-            var headerHolder = new HeaderHolder(configuration.getAliases());
-
-            AbstractFileReader<LineNumberReader> reader = new DefaultFileReader.Builder<LineNumberReader>(
-                    configuration.getCommon(),
-                    directoryChecker,
-                    new CsvContentParser(configuration.getAliases()),
-                    new MovedFileTracker(configuration.getSourceDirectory()),
-                    new InMemoryReaderState(),
-                    streamId -> commonFactory.newMessageIDBuilder().build(),
-                    Main::createSource
-            )
-                    .readFileImmediately()
-                    .acceptNewerFiles()
-                    .onSourceFound((streamId, path) -> clearHeader(headerHolder, streamId))
-                    .onContentRead((streamId, path, builders) -> attachHeaderOrHold(headerHolder, streamId, builders, configuration))
-                    .onStreamData((streamId, builders) -> publishMessages(rawMessageBatchRouter, streamId, builders))
-                    .onError((streamId, message, ex) -> publishErrorEvent(eventBatchRouter, streamId, message, ex, rootId))
-                    .onSourceCorrupted((streamId, path, e) -> publishSourceCorruptedEvent(eventBatchRouter, path, streamId, e, rootId))
-                    .build();
-            toDispose.add(reader);
+            final Runnable readerProcessUpdates;
```

Comment:
I think it is better to store the whole `AbstractFileReader` instead of a method reference to it